src/KurrentDB.SecondaryIndexing/Indexes/Default/DefaultIndexReader.cs

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Kurrent, Inc and/or licensed to Kurrent, Inc under one or more agreements.
+// Kurrent, Inc licenses this file to you under the Kurrent License v1 (see LICENSE.md).
+
+using KurrentDB.Core.Services.Storage.ReaderIndex;
+using KurrentDB.SecondaryIndexing.Storage;
+
+namespace KurrentDB.SecondaryIndexing.Indexes.Default;
+
+internal class DefaultIndexReader(
+	DuckDbDataSource db,
+	DefaultIndexProcessor processor,
+	DefaultIndexInFlightRecords inFlightRecords,
+	IReadIndex<string> index
+) : SecondaryIndexReaderBase(index) {
+	protected override long GetId(string streamName) => 0;
+
+	protected override long GetLastIndexedSequence(long id) => processor.LastSequence;
+
+	protected override IEnumerable<IndexedPrepare> GetIndexRecords(long _, long fromEventNumber, long toEventNumber) {
+		var range = db.Pool.Query<(long, long), AllRecord, DefaultSql.DefaultIndexQuery>((fromEventNumber, toEventNumber));
+		if (range.Count < toEventNumber - fromEventNumber + 1) {
+			var inFlight = inFlightRecords.TryGetInFlightRecords(fromEventNumber, toEventNumber);
+			range.AddRange(inFlight);
```

Comment:
what guarantees do we have that the in flight records and the db records are mutually exclusive? do the records need to be de-duplicated? (consider the case where the index is being read while the appender is being flushed)

Reply:
Because in-flight records get cleared on appender flush.